source/Calamari.AzureAppService/Behaviors/TargetDiscoveryBehaviour.cs

```diff
@@ -182,11 +175,16 @@ public static ServiceMessage CreateWebAppTargetCreationServiceMessage(string? re
                 parameters.Where(p => p.Value != null).ToDictionary(p => p.Key, p => p.Value!));
         }
 
-        public static AsyncPageable<GenericResource> GetResources(this SubscriptionResource subscription,
-            string resourceType, int pageSize, CancellationToken cancellationToken)
+        public static async Task<AzureResource[]> GetResourcesByType(this ArmClient armClient, params string[] types)
         {
-            return subscription.GetGenericResourcesAsync($"resourceType eq 'Microsoft.web/{resourceType}'",
-                top: pageSize, cancellationToken: cancellationToken);
+            var tenant = armClient.GetTenants().First();
+            var typeCondition = types.Any()
+                ? $"| where {string.Join(" or ", types.Select(t => $"type == '{t}'"))} |"
+                : string.Empty;
```

Comment:
I'd pull `string.Join(" or ", types.Select(t => $"type == '{t}'"))` into a new variable or two above this - it'll make it easier to see that the `| where XYZ |` clause is actually simple.
```suggestion
            var typesToRetrieveClause = string.Join(" or ", types.Select(t => $"type == '{t}'"));
            var typeCondition = types.Any()
                ? $"| where { typesToRetrieveClause } |"
                : string.Empty;
```

Reply:
Yep good call 👍